### PR TITLE
Feature/ds4 11 pick item from agv to assembly

### PR DESCRIPTION
**Pull request by MagnusHJensen**
I had to add test for AGV, and add the method.

But also add the method inside the assembly module, so the integration test passed.
Also added service test for assembly method I implemented.

### CODE REVIEW

File: backend/AGV/src/test/java/dk/sdu/controllab/agv/AGVIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.MOCK
+)
+@AutoConfigureMockMvc
+@TestPropertySource(
+    locations = "classpath:agv-integration-test.properties"
+)
+@EnableJpaRepositories(basePackages = "dk.sdu.controllab")
+@Transactional
+public class AGVIntegrationTest {
```
Comment (GustavFrandsen) [inline backend/AGV/src/test/java/dk/sdu/controllab/agv/AGVIntegrationTest.java:35]: when i try to run this code it fails misserably, if it is on purpose then just resolve this comment

Comment (MagnusHJensen) [reply backend/AGV/src/test/java/dk/sdu/controllab/agv/AGVIntegrationTest.java:35]: You might need to create a seperate database in postgres called “integration_test”

Comment (GustavFrandsen) [reply backend/AGV/src/test/java/dk/sdu/controllab/agv/AGVIntegrationTest.java:35]: could that maybe be added automatically?

Comment (MagnusHJensen) [reply backend/AGV/src/test/java/dk/sdu/controllab/agv/AGVIntegrationTest.java:35]: Unfortunately not, it should just be a one time thing to add it.
Unless you spin up a new database.

Comment (GustavFrandsen) [reply backend/AGV/src/test/java/dk/sdu/controllab/agv/AGVIntegrationTest.java:35]: still fail just a different error

File: backend/AGV/src/test/java/dk/sdu/controllab/agv/service/AGVServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import dk.sdu.controllab.agv.model.IAGVRepository;
+import dk.sdu.controllab.common.Item;
+import dk.sdu.controllab.commonagv.AGVProgram;
+import dk.sdu.controllab.commonassembly.AssemblyService;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
```
Comment (GustavFrandsen) [inline backend/AGV/src/test/java/dk/sdu/controllab/agv/service/AGVServiceImplTest.java:13]:
```suggestion
```

File: backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
 import dk.sdu.controllab.agv.model.IAGVRepository;
 import dk.sdu.controllab.commonagv.AGVProgram;
 import dk.sdu.controllab.commonagv.AGVState;
 import dk.sdu.controllab.commonagv.AGVStatusResponse;
 import dk.sdu.controllab.commonagv.IAGVService;
+import dk.sdu.controllab.commonassembly.AssemblyService;
 import io.swagger.client.ApiException;
 import io.swagger.client.api.DefaultApi;
 import io.swagger.client.model.LoadProgramRequest;
 import org.springframework.beans.factory.annotation.Autowired;
```
Comment (GustavFrandsen) [inline backend/AGV/src/main/java/dk/sdu/controllab/agv/service/AGVServiceImpl.java:14]:
```suggestion

```

**Review by GustavFrandsen — APPROVED** [review]
it works

**Pull request merged**